src/test/scala/BagSpec.scala

```diff
@@ -0,0 +1,10 @@
+import org.scalatest._
+
+class BagSpec extends FlatSpec with Matchers {
+
+  "initialize()" should "generate a random List of 101 Tiles" in {
+    Bag.initialise() should have length (101)
+    // TODO test for randomness
```

Comment:
I don't think you need to handle randomness in your test, see my comment about pushing the non determinism to the edge.  One approach is to pass in a "randomise" function, and then check that it's been called correctly.  But in the real code you would just pass in the actual Random.shuffle wheras in the test you can pass in a test function.